app/notification.go

```diff
@@ -649,32 +649,39 @@ func (a *App) sendPushNotification(post *model.Post, user *model.User, channel *
 	return nil
 }
 
-func (a *App) ClearPushNotification(userId string, channelId string) *model.AppError {
-	sessions, err := a.getMobileAppSessions(userId)
-	if err != nil {
-		return err
-	}
-
-	msg := model.PushNotification{}
-	msg.Type = model.PUSH_TYPE_CLEAR
-	msg.ChannelId = channelId
-	msg.ContentAvailable = 0
-	if badge := <-a.Srv.Store.User().GetUnreadCount(userId); badge.Err != nil {
-		msg.Badge = 0
-		l4g.Error(utils.T("store.sql_user.get_unread_count.app_error"), userId, badge.Err)
-	} else {
-		msg.Badge = int(badge.Data.(int64))
-	}
-
-	l4g.Debug(utils.T("api.post.send_notifications_and_forget.clear_push_notification.debug"), msg.DeviceId, msg.ChannelId)
+func (a *App) ClearPushNotification(userId string, channelId string) {
+	go func() {
```

Comment:
Just checking, but we're not worried about this being unbounded and backing up the number of goroutines?

Reply:
It doesn't happen very often because `ClearPushNotification` is only called when the current channel went to zero unreads and it was non-zero before.